tests/core/search/test_documents.py

```diff
@@ -0,0 +1,162 @@
+from core.search.document import (
+    BASIC_TEXT,
+    BINARY,
+    BOOLEAN,
+    BYTE,
+    CUSTOM_KEYWORD,
+    DOUBLE,
+    FILTERABLE_TEXT,
+    FLOAT,
+    INTEGER,
+    IP,
+    LONG,
+    SHORT,
+    UNSIGNED_LONG,
+    SearchMappingDocument,
+    date,
+    icu_collation_keyword,
+    keyword,
+    nested,
+    sort_author_keyword,
+)
+
+
+class TestDocuments:
+    def test_binary(self):
```

Comment:
Being a pedant again: All the type checking tests could be a single parameterized test

Reply:
Is there any benefit to doing this?

Reply:
Reusability if you want to make a change in the test code everywhere